independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ArcContainerImpl.java

```diff
@@ -291,12 +291,12 @@ public <X> InstanceHandle<X> instance(Type type, Annotation... qualifiers) {
 
     @Override
     public <T> Supplier<InstanceHandle<T>> beanInstanceSupplier(Class<T> type, Annotation... qualifiers) {
-        return createInstanceSupplier(false, type, qualifiers);
+        return createInstanceSupplier(true, type, qualifiers);
```

Comment:
This was an unfortunate bug from the start introduced by [this commit](https://github.com/quarkusio/quarkus/commit/c5b87cf365729d0ecfc60da5e415174978ae87a4#diff-06bb4f2eae4445212128b4f566fc42949f578c01b721589280cf9991a59ffb13).
`beanInstanceSupplier` is the variant that *should* resolve ambiguities while  `instanceSupplier` *should not*.